app/views/govuk_publishing_components/components/_google_tag_manager_script.html.erb

```diff
@@ -1,13 +1,19 @@
 <%
   gtm_auth ||= nil
   gtm_preview ||= nil
+  blocklist ||= nil # an array of the form ['customPixels', 'customScripts', 'html', 'nonGoogleScripts']
 
   gtm_attributes = %w(gtm_cookies_win=x)
     gtm_attributes << "gtm_auth=" + gtm_auth if gtm_auth
     gtm_attributes << "gtm_preview=" + gtm_preview if gtm_preview
   gtm_attributes = gtm_attributes.join('&')
 %>
+
 <script>
+<% if blocklist %>
+window.dataLayer = window.dataLayer || []
+window.dataLayer.push({ 'gtm.blocklist' : <%= blocklist %> })
```

Comment:
This seems to treat the variable as a string? whereas I'd expect it to be dealing with an array?

Reply:
Thanks @kevindew - I would expect that it depends on what `blocklist` is set to. The expectation is `# an array of the form ['customPixels', 'customScripts', 'html', 'nonGoogleScripts']` as per the comment. But without TypeScript...!